src/main/java/com/palantir/gradle/shadowjar/ComposableManifestAppenderTransformer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.shadowjar;
+
+import com.github.jengelman.gradle.plugins.shadow.transformers.ResourceTransformer;
+import com.github.jengelman.gradle.plugins.shadow.transformers.TransformerContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.jar.JarFile;
+import org.apache.tools.zip.ZipEntry;
+import org.apache.tools.zip.ZipOutputStream;
+import org.gradle.api.file.FileTreeElement;
+import org.gradle.api.tasks.Input;
+
+// Originally taken from https://github.com/GradleUp/shadow/blob/9.2.2/src/main/kotlin/com/github/jengelman/gradle/
```

Comment:
not sure if this link is still valid ? I am getting a 404 accessing this link 

the link below works:
https://github.com/GradleUp/shadow/blob/cba083d489446eb23fda25a7bfa59e3af93e32b1/src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/transformers/ManifestAppenderTransformer.kt